subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/PropertyValidationAccess.java

```diff
@@ -54,35 +54,26 @@
         InputDirectory.class, new MissingPathSensitivityValidator()
     );
 
+    private static final NestedBeanResolver<TypeNode> NESTED_BEAN_RESOLVER = new NestedBeanResolver<TypeNode>();
+
     @SuppressWarnings("unused")
     public static void collectTaskValidationProblems(Class<?> topLevelBean, Map<String, Boolean> problems) {
         DefaultTaskClassInfoStore taskClassInfoStore = new DefaultTaskClassInfoStore();
         PropertyMetadataStore metadataStore = new DefaultPropertyMetadataStore(ImmutableList.of(
             new ClasspathPropertyAnnotationHandler(), new CompileClasspathPropertyAnnotationHandler()
         ));
         Queue<TypeNode> queue = new ArrayDeque<TypeNode>();
-        queue.add(new TypeNode(null, TypeToken.of(topLevelBean)));
+        queue.add(new TypeNode(null, TypeToken.of(topLevelBean), metadataStore.getTypeMetadata(topLevelBean)));
```

Comment:
Why store the metadata instead of looking it up when needed? We already have `getTypeMetadata()` cached.

Reply:
This is not stored as part of the node any more.